test/util/Generators.scala

```diff
@@ -123,16 +111,17 @@ trait Generators {
     updatedByTeam = updatedByTeam
   )
 
-  val protectedUserRecordGen: Gen[ProtectedUserRecord] = for {
-    entryId      <- nonEmptyStringGen
-    firstCreated <- Gen.posNum[Long]
-    lastUpdated  <- Gen.option(Gen.posNum[Long])
-    body         <- protectedUserGen
-
-  } yield ProtectedUserRecord(
-    entryId = entryId,
-    firstCreated = firstCreated,
-    lastUpdated = lastUpdated,
-    body = body
+  implicit val arbProtectedUserRecord: Arbitrary[ProtectedUserRecord] = Arbitrary(
```

Comment:
I think we should avoid relying on implicit Arbitrary and stick with the explicit Gen in the forAll()  - using the implicit can make it harder to find the source of the generated data.  
Also as soon as you need a generator of the same type but with different fields it seems that you need to be clever about your import location so as to get the correct arb or you end up ditching the arb and using a generator anyway

Reply:
I think we need to consider the semantics of type classes here. The whole point of a type class is to define some behaviour which is universally true for a given type. As soon as a type has multiple type classes, there is a problem with the logic - one lane has to be picked and stuck with.

In particular, this makes sense when you consider the meaning of the word "arbitrary" - i.e. totally unbiased random pick of some given valid. If need generators for specific subsets of that type, e.g. valid/invalid values, they must be implemented explicitly as they are no longer arbitrary.

With this in mind, I feel I used the type class appropriately here. None of the tests drawing from the arbitrary instance care whether the values are valid or not - just that they are of the correct type.

Reply:
whilst that may be theoretically correct, my point is that it makes the code a little less easy/convenient to work with as its not immediately obvious where the the code that creates the arbitrary live, you need to hunt for it, whereas with an explicit Gen in the forAll its simple to drill into it using the IDE - in the real world this makes it a bit more practical

Reply:
I think I see the point about it being truly arbitrary and so the implicit giving you the general solution would be appropriate, and if a more restricted override is needed then an explicit Gen could be used, however I don't see the maintainability benefit of taking away the named explicit reference which a developer can read and potentially click through to the source. You'd have to look at the type, know/remember it's using an implicit somewhere and look for that which does feel like an avoidable loss of simplicity/clarity so on balance I'd prefer the explicit Gen. (As a general rule I'd always prefer 5 lines of clarity over 3 lines of wizardry 😆 )

Reply:
The benefit is that the existence of an `Arbitrary[ProtectedUserRecord]` automatically induces the existence of other useful generators, such as `Arbitrary[Option[ProtectedUserRecord]]`, `Arbitrary[Seq[ProtectedUserRecord]]`, and many others which take a `ProtectedUserRecord` as a type parameter with explicitly having to define a new generator for them.